src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -1238,7 +1244,8 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
         let config = config |> AVal.force
 
         match forceFindOpenFileOrRead filePath with
-        | Ok(fileInfo) -> return! parseAndCheckFile checker fileInfo opts config |> Async.Ignore
+        // Don't cache for autocompletions as we really only want to cache "Opened" files.
+        | Ok(fileInfo) -> return! parseAndCheckFile checker fileInfo opts config false |> Async.Ignore
```

Comment:
As the comment says, not work caching non opened files as you're not getting autocompletions for them.